Templates/CSharp/Model/ComplexType.cs.tt

```diff
@@ -43,10 +43,8 @@ namespace <#=complex.Namespace.GetNamespaceName()#>
     public <#=classType#> <#=typeDeclaration#>
     {
 <#
-// Generate a constructor to initialize the @odata.type property when this type is not abstract and if this 
-// type's base is abstract and the base is referenced as the type of a structural property. We need this
-// to disambiguate the type of the descendant class being sent. 
-if (complex.IsBaseAbstractAndReferencedAsPropertyType() && !complex.IsAbstract)
+// Generate a constructor to initialize the @odata.type property when this type is not abstract.
+if (!complex.IsAbstract)
```

Comment:
Should this be `if (complex.Base != null)`?

Reply:
I don't think it has to be. I can't recall why having this is generally useful.

Reply:
It would work both ways, I was just wondering.